istioctl/pkg/auth/upgrader.go

```diff
@@ -214,24 +214,51 @@ func (ug *Upgrader) convertClusterRbacConfig(authzPolicies *model.AuthorizationP
 	if clusterRbacConfig == nil {
 		return fmt.Errorf("no ClusterRbacConfig found")
 	}
+	// RbacConfig_ON_WITH_INCLUSION doesn't require the config root namespace.
+	if clusterRbacConfig.Mode == rbac_v1alpha1.RbacConfig_ON_WITH_INCLUSION {
+		// Support namespace-level only.
+		// For each namespace in RbacConfig_ON_WITH_INCLUSION, we simply generate a deny-all rule for that namespace.
+		return ug.generateClusterRbacConfig(rbacNamespaceDeny, clusterRbacConfig.Inclusion.Namespaces, false)
+	}
 	rootNamespace, err := ug.getIstioRootNamespace()
 	if err != nil {
 		return fmt.Errorf("failed to get Istio root namespace: %s", err)
 	}
-	rootNs := map[string]string{
-		"RootNamespace": rootNamespace,
-	}
-	var policy string
 	switch clusterRbacConfig.Mode {
 	case rbac_v1alpha1.RbacConfig_OFF:
-		policy, err = fillTemplate(rbacGlobalAllowAll, rootNs)
+		return ug.generateClusterRbacConfig(rbacNamespaceAllow, []string{rootNamespace}, true)
 	case rbac_v1alpha1.RbacConfig_ON:
-		policy, err = fillTemplate(rbacGlobalDenyAll, rootNs)
+		return ug.generateClusterRbacConfig(rbacNamespaceDeny, []string{rootNamespace}, true)
+	case rbac_v1alpha1.RbacConfig_ON_WITH_EXCLUSION:
+		// Support namespace-level only.
+		// First generate a cluster-wide deny rule.
+		err = ug.generateClusterRbacConfig(rbacNamespaceDeny, []string{rootNamespace}, true)
+		if err != nil {
+			return fmt.Errorf("failed to convert ClusterRbacConfig: %v", err)
+		}
+		// For each namespace in RbacConfig_ON_WITH_EXCLUSION, we simply generate an allow-rule rule for that namespace.
+		return ug.generateClusterRbacConfig(rbacNamespaceAllow, clusterRbacConfig.Exclusion.Namespaces, false)
 	}
-	if err != nil {
-		return fmt.Errorf("failed to convert ClusterRbacConfig: %v", err)
+	return nil
```

Comment:
Also could you error on the unsupported input? e.g. `RbacConfig_ON_WITH_INCLUSION` with services?

Reply:
Will do.